File: JavaScript/gnome_sort.js
function gnomeSort(arr) {
    let index = 0;
    while (index < arr.length) {
        if (index === 0 || arr[index] >= arr[index - 1]) {
            index++;
        } else {
            [arr[index], arr[index - 1]] = [arr[index - 1], arr[index]];
            index--;
        }
    }
    return arr;
}

const array = [34, 2, 10, -9, 0];
console.log(gnomeSort(array));

File: Kotlin/gnomeSort.kt
fun gnomeSort(arr: IntArray) {
    var index = 0
    while (index < arr.size) {
        if (index == 0 || arr[index] >= arr[index - 1]) {
            index++
        } else {
            val temp = arr[index]
            arr[index] = arr[index - 1]
            arr[index - 1] = temp
            index--
        }
    }
}

fun main() {
    val arr = intArrayOf(34, 2, 10, -9, 7, 0)
    gnomeSort(arr)
    println("Sorted array: ${arr.joinToString(", ")}")
}